src/linux/backend-v4l2.cpp

```diff
@@ -546,6 +546,63 @@ namespace librealsense
             }
         }
 
+        bool get_devname_from_video_path(const std::string& video_path, std::string& dev_name)
+        {
+            std::ifstream uevent_file(video_path + "/uevent");
+            if (!uevent_file)
+            {
+                LOG_ERROR("Cannot access " + video_path + "/uevent");
+                return false;
+            }
+
+            std::string uevent_line, major_string, minor_string;
+            while (std::getline(uevent_file, uevent_line) && (major_string.empty() || minor_string.empty()))
+            {
+                if (uevent_line.find("MAJOR=") != std::string::npos)
+                {
+                    major_string = uevent_line.substr(uevent_line.find_last_of('=') + 1);
+                }
+                else if (uevent_line.find("MINOR=") != std::string::npos)
+                {
+                    minor_string = uevent_line.substr(uevent_line.find_last_of('=') + 1);
+                }
+            }
+            uevent_file.close();
+
+            if (major_string.empty() || minor_string.empty())
+            {
+                LOG_ERROR("No Major or Minor number found for " + video_path);
+                return false;
+            }
+
+            DIR * dir = opendir("/dev");
+            if (!dir)
+            {
+                LOG_ERROR("Cannot access /dev");
+                return false;
+            }
+            while (dirent * entry = readdir(dir))
+            {
+                std::string name = entry->d_name;
+                std::string path = "/dev/" + name;
+
+                struct stat st = {};
+                if (stat(path.c_str(), &st) < 0)
+                {
+                    continue;
+                }
+
+                if (std::stoi(major_string) == major(st.st_rdev) && std::stoi(minor_string) == minor(st.st_rdev))
+                {
+                    dev_name = path;
+                    break;
+                }
+            }
+            closedir(dir);
+
+            return true;
```

Comment:
This method will return true if character device is not found in above while loop,
the only way to understand that it was not populated is to check if "dev_name" is empty?
Example - you have sysfs entry with video0/uevent 81:0 but there is no device node in container /dev/video{?}.
I think this was not the intention of this function.

Reply:
Yes good catch, I'll fix that now